sopel/modules/tld.py

```diff
@@ -234,8 +234,20 @@ def _update_tld_data(bot, which):
         bot.memory['tld_list_cache_updated'] = now
     elif which == 'data':
         try:
-            tld_data = requests.get(WIKI_PAGE_URI).text
-        except requests.exceptions.RequestException:
+            # https://www.mediawiki.org/wiki/Special:MyLanguage/API:Get_the_contents_of_a_page
+            tld_response = requests.get(
+                "https://en.wikipedia.org/w/api.php",
+                params={
+                    "action": "parse",
+                    "format": "json",
+                    "prop": "text",
+                    "utf8": 1,
+                    "formatversion": 2,
+                    "page": WIKI_PAGE_NAME,
+                },
+            ).json()
+            tld_data = tld_response["parse"]["text"]
+        except Exception:
```

Comment:
Don't really like that change. Any exception that isn't from Requests is very unexpected and should fail loudly instead of just logging a warning.

I guess in addition to `RequestException`s, you could also catch ~~`JSONDecodeError`~~ (not available in py2) `ValueError` just in case decoding the JSON fails for some reason? But any other kind of error really shouldn't get swept under the `except Exception` rug. You'll find a number of `# TODO: Be specific` comments throughout Sopel's code because we're trying to eliminate this pattern wherever possible.

Reply:
I can't think of anything but syntax errors, requests errors, JSONDecodeError, or KeyError. What would cause ValueError?
Still, fair.

Reply:
In older Pythons, `json` throws `ValueError` if decoding the JSON document fails. In newer Pythons, it throws `JSONDecodeError`—a subclass of `ValueError`. So, catching `ValueError` is valid to catch failed JSON decoding across all Python versions Sopel supports in 7.x.